types/chartist/chartist-tests.ts

```diff
@@ -563,4 +543,247 @@ new Chartist.Line('.ct-chart', {
           ]
         }
       ]
-}, {})
+}, {});
+
+var path: string = Chartist.Svg.Path.join(
+    [
+        new Chartist.Svg.Path()
+            .move(5, 5)
+            .line(10, 10)
+            .curve(10, 10, 0, 0, -5, 5, false, { curvy: 'indeed' })
+            .scale(10, 10)
+            .transform(function(pathElement, paramName, pathElementIndex, paramIndex, pathElements) {
+                if (pathElement.command === 'm') {
+                    pathElement.x += 1;
+                    pathElement.y -= 2;
+                } else if (pathElement.command === 'a') {
+                    pathElement.lAf = pathElement.rx = pathElement.xAr = 500;
+                }
+            }),
+        new Chartist.Svg.Path(),
+    ],
+    true,
+    {
+        accuracy: 3,
+    }
+).stringify();
+
+// --- Chartist.extend()
+
+var extend1: string = Chartist.extend({ a: 1 }, { a: 'one' }).a;
+
+// Expect:
+//     Type 'string' is not assignable to type 'number'
+// $ExpectError
+var extend2: number = Chartist.extend({ a: 1 }, { a: 'one' }).a;
+
+var extend3: string = Chartist.extend({ a: 1 }, { a: 0 as any }).a;
+
+var extend4: number = Chartist.extend({ a: 1 }, {}).a;
+
+// Note: This behavior of Chartist.extend() differs from Object.assign()
+var extend5: undefined = Chartist.extend({ a: 1 }, { a: undefined }).a;
+
+// Expect:
+//     Type 'number' is not assignable to type 'string'
+// $ExpectError
+var extend6: string = Chartist.extend({ a: 1 }, { a: {} }).a;
+
+// Expect:
+//     Argument of type '"hi"' is not assignable to parameter of type 'object'
+// $ExpectError
+Chartist.extend({ a: 1, b: 2 }, "hi").toLowerCase(); // extend7
+
+// Pass
+var extend8: number = Chartist.extend({ a: 1, b: 2 }, [1, 2, 3])[0];
+
+// Expect:
+//     Type 'number' is not assignable to type 'string'
+// $ExpectError
+var extend9: string = Chartist.extend({ a: 1, b: 2 }, [1, 2, 3])[0];
+
+var extend10: { a: { b: { c: string; d: number } }; e: number } = Chartist.extend(
+    { a: { b: { c: 'hi' } } },
+    { a: { b: { d: 1 } }, e: 2 }
+);
+
+// --- Plugin API
+
+Chartist.plugins = Chartist.plugins || {};
+
+Chartist.plugins.fakePlugin = function(options) {
+    return function(chart: Chartist.IChartistBase<any, any>) {};
+};
+
+Chartist.plugins.someOtherPlugin(123)(
+    new Chartist.Bar(
+        '.bar',
+        {
+            series: [],
+            labels: [],
+        },
+        {
+            plugins: [Chartist.plugins.superCharts()],
+        }
+    )
+);
+
+// -- SVG API
+
+var svgNode: SVGGElement = document.createElementNS('http://www.w3.org/2000/svg', 'g');
+var chartistSvgGroup: Chartist.IChartistSvg<SVGGElement> = new Chartist.Svg(svgNode);
+var charistSvgLine: Chartist.IChartistSvg<SVGLineElement> | null = chartistSvgGroup.querySelector('line');
+
+// --- Axis API
```

Comment:
This is dumped from real run-time axis data, with the exception of the functions, which are placeholders. Nonetheless, the purpose of this is to ensure our types correctly match real-world Chartist data.